torch/distributed/_tensor/placement_types.py

```diff
@@ -367,14 +366,23 @@ def __str__(self) -> str:
         return "P"
 
 
+class TensorMeta(NamedTuple):
```

Comment:
Not a strong opinion here, maybe call it `DTensorMeta`? WDYT?

Reply:
I think for this dataclass is intended to record "tensor" metadata like shape/stride/dtype, these fields are not quite related to `DTensor`, if we rename it to "DTensorMeta" I think we would expect this to include "device_mesh" and "placements", which is already recorded in the DTensorSpec, this might lead to some confusion. I'll also think more about the naming and see whether we need to consolidate it with DTensorSpec a bit